packages/idyll-components/src/map.js

```diff
@@ -0,0 +1,48 @@
+const React = require('react');
+const { mapChildren } = require('idyll-component-children');
+import TextContainer from './text-container';
+
+class Map extends React.Component {
+  render() {
+    const { children, value, currentValue } = this.props;
+
+    if (children) {
+      return mapChildren(children, child => {
+        return value.map(val => {
+          let newProps = Object.assign({}, child.props);
+          newProps = Object.keys(child.props).reduce((props, elm) => {
+            if (props[elm] === currentValue) {
+              props[elm] = val;
+              return props;
+            }
```

Comment:
Let's remove the references to `currentValue` and just have a hard coded keyword that will be replaced by the mapper, say, `_iterator`.

That way

```
[Map value:`['one', 'two', 'three]`]
  [Display value:'_iterator' /]
[/Map]
```

will work, and I think its a little more straightforward.

Reply:
Updated this as per your comment.